org.osgi.test.cases.typedevent/src/org/osgi/test/cases/typedevent/junit/TypedEventCapabilityTestCase.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 
+ *******************************************************************************/
+
+package org.osgi.test.cases.typedevent.junit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.osgi.namespace.implementation.ImplementationNamespace.*;
+import static org.osgi.namespace.service.ServiceNamespace.*;
+import static org.osgi.resource.Namespace.CAPABILITY_USES_DIRECTIVE;
+import static org.osgi.service.typedevent.TypedEventConstants.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.Version;
+import org.osgi.framework.wiring.BundleCapability;
+import org.osgi.framework.wiring.BundleWiring;
+import org.osgi.service.typedevent.TypedEventBus;
+import org.osgi.service.typedevent.monitor.TypedEventMonitor;
+import org.osgi.test.common.annotation.InjectBundleContext;
+import org.osgi.test.junit5.context.BundleContextExtension;
+import org.osgi.test.junit5.service.ServiceExtension;
+
+@ExtendWith(BundleContextExtension.class)
+@ExtendWith(ServiceExtension.class)
+@Nested
+public class TypedEventCapabilityTestCase {
+
+	@InjectBundleContext
+	BundleContext ctx;
+
+	@SuppressWarnings("unchecked")
+	private void testServiceCapability(Class< ? > clz) {
+
+		String className = clz.getName();
+
+		ServiceReference< ? > ref = ctx.getServiceReference(clz);
+
+		assertThat(ref).isNotNull();
+
+		BundleWiring wiring = ref.getBundle().adapt(BundleWiring.class);
+
+		BundleCapability cap = wiring.getCapabilities(SERVICE_NAMESPACE)
+				.stream()
+				.filter(c -> ((List<String>) c.getAttributes()
+						.get(CAPABILITY_OBJECTCLASS_ATTRIBUTE))
+								.contains(className))
+				.findFirst()
+				.orElse(null);
+
+		assertThat(cap).isNotNull();
+
+		assertThat(cap.getDirectives().get(CAPABILITY_USES_DIRECTIVE))
+				.isEqualTo(clz.getPackage().getName());
+	}
+
+	@Test
+	public void test_bus_service_capability() {
+		testServiceCapability(TypedEventBus.class);
```

Comment:
You should instead consider using parameterized tests.

https://github.com/osgi/osgi/blob/b618b4b9162d057acf46aceb903a52642d57c245/org.osgi.test.cases.typedevent/src/org/osgi/test/cases/typedevent/junit/TypedEventBusTest.java#L108-L118